src/parser/core/modules/items/bfa/enchants/Navigation.js

```diff
@@ -0,0 +1,163 @@
+    return Object.values(items).some(item => this.itemHasTrackedEnchant(item));
+  }
+  on_byPlayer_changebuffstack(event) {
+    if (event.ability.guid !== this.smallBuffToTrack) {
+      return;
+    }
+
+    this.buffStacks[event.start] = event.stackHistory;
+  }
+
+  get cleanStacks() {
+    const cleanStacks = {
+      0: [
+        {
+          start: this.owner.fight.start_time,
+        },
+      ],
+    };
+    let lastHandledStack = 0;
+
+    Object.values(this.buffStacks).forEach((stackChain) => {
+      stackChain.forEach((stack) => {
+        let stackSize = stack.stacks;
+        if (stackSize === 5){
+          stackSize = 0;
+        }
+        const stackStart = stack.timestamp;
+
+        if (cleanStacks.hasOwnProperty(lastHandledStack)) {
+          const previousStack = cleanStacks[lastHandledStack];
+          const lastOccurrence = previousStack[previousStack.length - 1];
+          if (!(lastOccurrence.hasOwnProperty('end'))) {
+            lastOccurrence.end = stackStart;
+            lastOccurrence.duration = lastOccurrence.end - lastOccurrence.start;
+          }
+        }
+
+        if (!(cleanStacks.hasOwnProperty(stackSize))) {
+          cleanStacks[stackSize] = [];
+        }
+
+        const stackInfo = {
+          start: stackStart,
+        };
+        cleanStacks[stackSize].push(stackInfo);
+        lastHandledStack = stackSize;
+      });
+    });
+
+    if (cleanStacks.hasOwnProperty(lastHandledStack)) {
+      const previousStack = cleanStacks[lastHandledStack];
+      const lastOccurrence = previousStack[previousStack.length - 1];
+      if (!(lastOccurrence.hasOwnProperty('end'))) {
+        lastOccurrence.end = this.owner.fight.end_time;
+        lastOccurrence.duration = lastOccurrence.end - lastOccurrence.start;
+      }
+    }
+
+    return cleanStacks;
+  }
+  maxStackBuffUptime() {
+    return this.selectedCombatant.getBuffUptime(this.bigBuffToTrack);
+  }
+  get averageStat() {
+    let averageStacks = 0;
+    const buffStacks = this.cleanStacks;
+
+    Object.keys(buffStacks).forEach((stackSize) => {
+      let totalStackDuration = 0;
+      buffStacks[stackSize].forEach((occurrence) => {
+        totalStackDuration += occurrence.duration;
+      });
+      averageStacks += totalStackDuration / this.owner.fightDuration * stackSize;
+    });
+    const maxStackUptimePercentage = this.maxStackBuffUptime() / this.owner.fightDuration;
+    return (averageStacks * this.statPerStack + maxStackUptimePercentage * this.statAtMax).toFixed(2);
+  }
+  item() {
+    const buffStacks = this.cleanStacks;
+    const maxStackBuffDuration = this.maxStackBuffUptime();
+    const tooltipData = (
+      <ExpandableStatisticBox
+        icon={<SpellIcon id={this.smallBuffToTrack} />}
+        value={`${this.averageStat}`}
+        label={`average ${this.primairyStat} gained`}
+        category={STATISTIC_CATEGORY.ITEMS}
+      >
+        <table className="table table-condensed">
```

Comment:
Mainly followed `IntoTheFray.js`, and thought that was what #2198 was asking :S

Reply:
I meant just an average stat + uptime 😄 I think it's important we don't show data that can't be used since we're already showing sooo much that we need to focus on just the important things so other data that can't be used for any choice or info don't distract.